sdk/java-sdk/src/main/scala/com/akkaserverless/javasdk/impl/eventsourcedentity/EventSourcedEntitiesImpl.scala

```diff
@@ -121,15 +122,18 @@ final class EventSourcedEntitiesImpl(
           source.via(runEntity(init))
         case (Seq(), _) =>
           // if error during recovery in proxy the stream will be completed before init
-          log.warning("Event Sourced Entity stream closed before init.")
+          log.error("Event Sourced Entity stream closed before init.")
           Source.empty[EventSourcedStreamOut]
         case (Seq(EventSourcedStreamIn(other, _)), _) =>
           throw ProtocolException(
             s"Expected init message for Event Sourced Entity, but received [${other.getClass.getName}]")
       }
       .recover { case error =>
-        log.error(error, failureMessage(error))
-        EventSourcedStreamOut(OutFailure(failure(error)))
+        // only "unexpected" exceptions should end up here
+        ErrorHandling.withCorrelationId { correlationId =>
+          log.error(error, failureMessageForLog(error))
```

Comment:
I'm wondering if we should do the same trick as for actions for all kind of entities. We will need to init the handler outside the flow with minor penalty when we don't use it (case 2 and 3 in the pattern matching), but I think that case 2 and 3 are rather rare. So might be an option?

Reply:
We don't know which entity to init unless we have the init message with the service name in it, so no service to tie log to here.

Reply:
true, I missed that.